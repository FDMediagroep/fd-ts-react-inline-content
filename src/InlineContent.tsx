import {BulletPoints, BulletPointsStyle} from './components/BulletPoints';
import {LinkBlock, LinkBlockStyle} from './components/LinkBlock';
import {NumberFrame, NumberFrameStyle} from './components/NumberFrame';
import {Quote, QuoteStyle} from './components/Quote';
import {StackFrame, StackFrameStyle, StackFrameDataType} from './components/StackFrame';
import {Stock, StockStyle} from './components/Stock';
import {Summary, SummaryStyle} from './components/Summary';
import {TextFrame, TextFrameStyle} from './components/TextFrame';

export {BulletPoints, BulletPointsStyle};
export {LinkBlock, LinkBlockStyle};
export {NumberFrame, NumberFrameStyle};
export {Quote, QuoteStyle};
export {StackFrame, StackFrameStyle, StackFrameDataType};
export {Stock, StockStyle};
export {Summary, SummaryStyle};
export {TextFrame, TextFrameStyle};
