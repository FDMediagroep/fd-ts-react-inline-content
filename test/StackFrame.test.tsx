import React from 'react';
import TestRenderer from 'react-test-renderer';
import { StackFrame } from '../src/InlineContent';

describe('StackFrame', ()  => {
    const stackFrames = [
        {title: 'Vivamus varius sed quam vel aliquam.', paragraph: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'},
        {title: 'Ut eleifend porta arcu, a eleifend enim viverra sit amet.', paragraph: 'Quisque a commodo eros. Sed at mi consequat, lacinia eros sed, luctus velit.'}
    ];

    it('should render align right correctly', () => {
        let component = TestRenderer.create(<StackFrame
            stackFrames={stackFrames}
        />);
        expect(component.toJSON()).toMatchSnapshot();

        component = TestRenderer.create(<StackFrame
            stackFrames={stackFrames}
            align='right'
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render align block correctly', () => {
        let component = TestRenderer.create(<StackFrame
            stackFrames={stackFrames}
            align='block'
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });

});
