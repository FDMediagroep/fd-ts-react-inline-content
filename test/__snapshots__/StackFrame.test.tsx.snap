// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackFrame should render align block correctly 1`] = `
<div
  className="inline-content stapelkader block"
>
  <h3
    className="fd-typography article-h-inline-stack-frame"
    textStyle="article-h-inline-stack-frame"
  >
    Vivamus varius sed quam vel aliquam.
  </h3>
  <p
    className="fd-typography article-p-inline-stack-frame"
    textStyle="article-p-inline-stack-frame"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>
  <h3
    className="fd-typography article-h-inline-stack-frame"
    textStyle="article-h-inline-stack-frame"
  >
    Ut eleifend porta arcu, a eleifend enim viverra sit amet.
  </h3>
  <p
    className="fd-typography article-p-inline-stack-frame"
    textStyle="article-p-inline-stack-frame"
  >
    Quisque a commodo eros. Sed at mi consequat, lacinia eros sed, luctus velit.
  </p>
</div>
`;

exports[`StackFrame should render align right correctly 1`] = `
<div
  className="inline-content stapelkader right"
>
  <h3
    className="fd-typography article-h-inline-stack-frame"
    textStyle="article-h-inline-stack-frame"
  >
    Vivamus varius sed quam vel aliquam.
  </h3>
  <p
    className="fd-typography article-p-inline-stack-frame"
    textStyle="article-p-inline-stack-frame"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>
  <h3
    className="fd-typography article-h-inline-stack-frame"
    textStyle="article-h-inline-stack-frame"
  >
    Ut eleifend porta arcu, a eleifend enim viverra sit amet.
  </h3>
  <p
    className="fd-typography article-p-inline-stack-frame"
    textStyle="article-p-inline-stack-frame"
  >
    Quisque a commodo eros. Sed at mi consequat, lacinia eros sed, luctus velit.
  </p>
</div>
`;

exports[`StackFrame should render align right correctly 2`] = `
<div
  className="inline-content stapelkader right"
>
  <h3
    className="fd-typography article-h-inline-stack-frame"
    textStyle="article-h-inline-stack-frame"
  >
    Vivamus varius sed quam vel aliquam.
  </h3>
  <p
    className="fd-typography article-p-inline-stack-frame"
    textStyle="article-p-inline-stack-frame"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>
  <h3
    className="fd-typography article-h-inline-stack-frame"
    textStyle="article-h-inline-stack-frame"
  >
    Ut eleifend porta arcu, a eleifend enim viverra sit amet.
  </h3>
  <p
    className="fd-typography article-p-inline-stack-frame"
    textStyle="article-p-inline-stack-frame"
  >
    Quisque a commodo eros. Sed at mi consequat, lacinia eros sed, luctus velit.
  </p>
</div>
`;
